src/test/java/com/cloudbees/plugins/credentials/CredentialsProviderTest.java

```diff
@@ -300,11 +296,10 @@ public void testNodeCredentialFingerprintsAreRemovedForNonExistantNodes() throws
         CredentialsProvider.track(addedSlave, globalCred);
         assertEquals(initialFingerprintSize+1, CredentialsProvider.getOrCreateFingerprintOf(globalCred).getFacets().size());
 
-        // Remove the added slave from Jenkins, and register the
-        // slave again to flush any mapped credentials for nodes that no-longer
-        // exist - including this one
+        // Remove the added slave from Jenkins, and track the non-added slave
+        // to flush any mapped credentials for nodes that no longer exist.
         Jenkins.getInstance().removeNode(addedSlave);
-        CredentialsProvider.track(addedSlave, globalCred);
+        CredentialsProvider.track(nonAddedSlave, globalCred);
```

Comment:
If we track `addedSlave` then we never cover the following branch in the tests:

    } else if (!jenkinsNodeNames.contains(((NodeCredentialsFingerprintFacet) f).getNodeName())) {
        iterator.remove();
    }